tests/spcs/test_image_repository.py

```diff
@@ -140,43 +140,47 @@ def test_create_repository_already_exists(mock_handle, mock_execute):
     )
 
 
-@mock.patch("snowflake.cli._plugins.spcs.image_repository.commands.requests.get")
-@mock.patch(
-    "snowflake.cli._plugins.spcs.image_repository.commands.ImageRepositoryManager._execute_query"
+@patch(
+    "snowflake.cli._plugins.spcs.image_repository.commands.ImageRepositoryManager.list_images"
 )
-@mock.patch(
-    "snowflake.cli._plugins.spcs.image_repository.commands.ImageRepositoryManager._conn"
-)
-@mock.patch(
-    "snowflake.cli._plugins.spcs.image_registry.commands.RegistryManager.login_to_registry"
-)
-def test_list_images(
-    mock_login,
-    mock_conn,
-    mock_execute,
-    mock_get_images,
+def test_list_images_cli(
+    mock_list_images,
     runner,
     mock_cursor,
 ):
-    mock_conn.database = "DB"
-    mock_conn.schema = "SCHEMA"
-    mock_conn.role = "MY_ROLE"
-
-    mock_execute.return_value = mock_cursor(
-        rows=MOCK_ROWS_DICT,
-        columns=MOCK_COLUMNS,
+    cursor = mock_cursor(
+        rows=[
+            [
+                "2024-10-11 14:23:49-07:00",
+                "echo_service",
+                "latest",
+                "sha256:a8a001fef406fdb3125ce8e8bf9970c35af7084",
+                "/db/schema/repo/echo_service",
+            ]
+        ],
+        columns=["created_on", "image_name", "tags", "digest", "image_path"],
     )
-    mock_login.return_value = "TOKEN"
-
-    mock_get_images.return_value.status_code = 200
-    mock_get_images.return_value.text = '{"repositories":["baserepo/super-cool-repo"]}'
+    mock_list_images.return_value = cursor
 
     result = runner.invoke(
         ["spcs", "image-repository", "list-images", "IMAGES", "--format", "JSON"]
     )
 
     assert result.exit_code == 0, result.output
-    assert json.loads(result.output) == [{"image": "/DB/SCHEMA/IMAGES/super-cool-repo"}]
+    assert "/db/schema/repo/echo_service" in result.output
```

Comment:
nit: why don't we assert cursor == result directly?

Reply:
The output is in table format. I guess we could make it json and compare. But the return value is faked entirely. For example, it has the same effect as making `cursor` return `foo` and assert the result is the same as `foo`.